api/python/cellxgene_census/src/cellxgene_census/experimental/util/_csr_iter.py

```diff
@@ -81,42 +100,51 @@ def X_sparse_iter(
     obs_coords = query.obs_joinids().to_numpy()
     obs_coord_chunker = (obs_coords[i : i + stride] for i in range(0, len(obs_coords), stride))
 
-    # Lazy read into Arrow Table. Yields (coords, Arrow.Table)
     X = query._ms.X[X_name]
     var_coords = query.var_joinids().to_numpy()
+
+    # Lazy read of X chunk. Yields ((obs_coords, var_coords), (soma_data, soma_dim_0, soma_dim_1))
+    def soma_tbl_vectors(tbl: pa.Table) -> Tuple[npt.NDArray[np.float64], npt.NDArray[np.int64], npt.NDArray[np.int64]]:
+        return tuple(tbl[col].to_numpy() for col in ("soma_data", "soma_dim_0", "soma_dim_1"))  # type: ignore
+
     table_reader = (
         (
             (obs_coords_chunk, var_coords),
-            X.read(coords=(obs_coords_chunk, var_coords)).tables().concat(),
+            soma_tbl_vectors(X.read(coords=(obs_coords_chunk, var_coords)).tables().concat()),
         )
         for obs_coords_chunk in obs_coord_chunker
     )
     if use_eager_fetch:
         table_reader = (t for t in _EagerIterator(table_reader, query._threadpool))
 
-    # lazy reindex of obs coordinates. Yields coords and (data, i, j) as numpy ndarrays
-    coo_reindexer = (
-        (
-            (obs_coords_chunk, var_coords),
+    coo_reindexer = table_reader
+    if reindex_obs or reindex_var:
+        # lazy reindexing of soma_dim_0 and soma_dim_1 to obs_coords and var_coords positions
+        coo_reindexer = (
             (
-                tbl["soma_data"].to_numpy(),
-                pd.Index(obs_coords_chunk).get_indexer(tbl["soma_dim_0"].to_numpy()),
-                query.indexer.by_var(tbl["soma_dim_1"].to_numpy()),
-            ),
+                (obs_coords_chunk, var_coords),
+                (
+                    soma_data,
+                    pd.Index(obs_coords_chunk).get_indexer(soma_dim_0) if reindex_obs else soma_dim_0,
+                    query.indexer.by_var(soma_dim_1) if reindex_var else soma_dim_1,
+                ),
+            )
+            for (obs_coords_chunk, var_coords), (soma_data, soma_dim_0, soma_dim_1) in table_reader
         )
-        for (obs_coords_chunk, var_coords), tbl in table_reader
-    )
-    if use_eager_fetch:
-        coo_reindexer = (t for t in _EagerIterator(coo_reindexer, query._threadpool))
+        if use_eager_fetch:
+            coo_reindexer = (t for t in _EagerIterator(coo_reindexer, query._threadpool))
 
-    # lazy convert Arrow table to Scipy sparse.csr_matrix
+    # lazy convert to Scipy sparse.csr_matrix
```

Comment:
This comment is also wrong - it can convert to CSC _or_ CSR (or it used to do that).  The variable name is also misleading.   (yes, some of this is my fault :-)

Suggest clarifying - the `fmt_ctor` controls the format, via the user-provided `fmt` flag